src/app/[locale]/selected-filters.tsx

```diff
@@ -0,0 +1,114 @@
+import {SearchResultFilter} from '@/lib/dataset-fetcher';
+import {Dispatch} from 'react';
+import {useTranslations} from 'next-intl';
+import {XMarkIcon} from '@heroicons/react/24/solid';
+
+interface TagProps {
+  name?: string;
```

Comment:
Out of interest: in which case does a tag not have a `name` (since it's optional) but can be removed (since `remove` is required)?